src/context/context-container.ts

```
@@ -1,48 +1,64 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
 import { Context } from './context';
 import { Inject, Injectable } from '@nestjs/common';
 import { ModuleRef } from '@nestjs/core';
+import LRUCache from 'lru-cache';
 import { CONTEXT_MODULE_CONFIG, HEADER_REQUEST_ID } from '../constants';
 import { ContextConfigType, RequestType } from '../interfaces';
-import { generateId } from '../tools';
+import { ClsService } from 'nestjs-cls';
 
 @Injectable()
 export class ContextContainer {
   private contexts: Record<string, Context> = {};
-  private contextStack: string[] = [];
+  private cache: LRUCache<string, Context>;
 
```

The `id` recover process for the context should return an exception if we are not able to get it from the ClsService or the request. This exceptionb should be logged to prevent the user that the lib has been missused